test/extensions_and_deriving/test.ml

```diff
@@ -51,7 +51,7 @@ val gen_type_decl : Context_free.Rule.t = <abstr>
 
 [%%expect{|
 Line _, characters 3-19:
-Error: Attributes not allowed here
+Error: (ppx <builtin:context-free>) Attributes not allowed here
```

Comment:
This error is thrown by ppxlib. So in this case, pointing to a PPX (or saying the error was thrown while applying a PPX) might be confusing, since that suggests that the error is somehow related to the PPX, whereas, in fact, it's simply the PPX user who is writing an attribute at a place where the parsetree doesn't expect attributes.

I'm thinking about how we could fix this. It's just a manifestation of the following problem: all located errors raised by ppxlib when traversing the AST to apply all context-free rules (i.e. everything that happens in `Context_free.map_top_down`) are caught just in the same way as the located errors raised by PPXs are caught. I can see two options to approach this:
1. We could simply not improve the error message in the case of located errors inside context-free rules. Once we've approached your issue #291 on handling that kind of errors correctly in ppxlib, that shouldn't be a problem anymore anyways. That might be the most reasonable solution.
2. You could have a look into how complicated it would be to divide the logic of catching exceptions into those coming from whole-file transformations (catch them in `driver.ml` just as you do now) and those coming from rules (catch them in `Context_free.map_top_down`.

What do you think about this? And how do the two options, particularly option 1, sound to you?